src/main/java/minijava/parser/Parser.java

```diff
@@ -467,6 +468,8 @@ private void parseNewObjectArrayExpression() {
           case LBRACKET:
             parseNewArrayExpression();
             break;
+          default:
+            unexpectCurrentToken();
```

Comment:
Maybe change `expectAndConsume` to take var args, so that error messages can be better? e.g. `expectAndConsume(LBRACKET, LPAREN)`